engine/flink/kafka-util/src/main/scala/pl/touk/nussknacker/engine/kafka/source/KafkaGenericNodeSourceFactory.scala

```diff
@@ -0,0 +1,73 @@
+package pl.touk.nussknacker.engine.kafka.source
+
+import pl.touk.nussknacker.engine.flink.api.timestampwatermark.TimestampWatermarkHandler
+import org.apache.flink.types.Nothing
+import org.apache.kafka.clients.consumer.ConsumerRecord
+import pl.touk.nussknacker.engine.api.context.{ProcessCompilationError, ValidationContext}
+import pl.touk.nussknacker.engine.api.context.transformation.{NodeDependencyValue, SingleInputGenericNodeTransformation}
+import pl.touk.nussknacker.engine.api.definition.{DualParameterEditor, NodeDependency, Parameter, StringParameterEditor}
+import pl.touk.nussknacker.engine.api.editor.DualEditorMode
+import pl.touk.nussknacker.engine.api.process.ProcessObjectDependencies
+import pl.touk.nussknacker.engine.flink.api.process.{FlinkContextInitializer, FlinkSource}
+import pl.touk.nussknacker.engine.kafka.{KafkaConfig, KafkaUtils, RecordFormatter}
+import pl.touk.nussknacker.engine.kafka.serialization.KafkaDeserializationSchemaFactory
+
+import scala.reflect.ClassTag
+
+
+/**
+  * Base factory for Kafka sources with additional metadata variable.
+  * It is based on [[pl.touk.nussknacker.engine.api.context.transformation.SingleInputGenericNodeTransformation]]
+  * that allows custom ValidationContext and Context transformations, which are provided by [[pl.touk.nussknacker.engine.kafka.source.KafkaGenericContextInitializer]]
+  *
+  * @param deserializationSchemaFactory - produces KafkaDeserializationSchema for [[pl.touk.nussknacker.engine.kafka.source.KafkaSource]]
+  * @param timestampAssigner            - provides timestampAsigner and WatermarkStrategy to KafkaSource
+  * @param formatter                    - support for test data parser and generator
+  * @param processObjectDependencies
+  * @tparam K - type of key of kafka event that is generated by raw source (SourceFunction).
+  * @tparam V - type of value of kafka event that is generated by raw source (SourceFunction).
+  */
+class KafkaGenericNodeSourceFactory[K: ClassTag, V: ClassTag](deserializationSchemaFactory: KafkaDeserializationSchemaFactory[ConsumerRecord[K, V]],
```

Comment:
1. Why you declare here class tags? I can't find usage of them.
2. I don't like `genericnode` in this name.  IMO developers without background that we initially didn't have generic nodes and meantime they were introduced, will be confused why this class is more generic then some other. I have other options:
a) let's skip phasing out of "old" `KafkaSourceFactory` and call this just `KafkaSourceFactory`
b) let's call it `KafkaSourceFactory_v2`
c)  let's call it `KafkaSourceFactoryWithMeta` or sth similar to name purpose of this class instead how it is implemented.

Reply:
1. Class tags are required by `KafkaContextInitializer` to set proper `ValidationContext`
2. I've chosen option a), hence only one `KafkaSourceFactory`